### ISSUE TITLE
Index out of bounds error

### ISSUE THREAD

**Issue by olafurpg** [OlivierBlanvillain/monadic-html#16]
```scala
    val count: Var[Int] = Var[Int](0)

    val dogs: Rx[Seq[xml.Node]] =
      count.map(i =>
        Seq.fill(i)(
          <img style="width: 100px;" src="http://doge2048.com/meta/doge-600.png"></img>))
    count.filter(_ > 0).foreach(println)

    <div style="background-color: darkorange;">
      <button onclick={ () => count.update(_ + 1) }>Click Me!</button>
      {count.filter(_ > 0).map(<h2>WOW!!!</h2>)}
      {count.filter(_ > 2).map(<h2>MUCH REACTIVE!!!</h2>)}
      {count.filter(_ > 5).map(<h2>SUCH BINDING!!!</h2>)}
      {dogs}
    </div>
```

Produces the following error
```
hd	@	LinearSeqOptimized.scala:65
c.la	@	List.scala:84
c.j	@	NodeSeq.scala:55
(anonymous function)	@	rx.scala:14
Bo.j	@	AnonFunctions.scala:15
Qb	@	rx.scala:77
(anonymous function)	@	rx.scala:18
Bo.j	@	AnonFunctions.scala:15
Qb	@	rx.scala:77
(anonymous function)	@	rx.scala:83
Bo.j	@	AnonFunctions.scala:15
Lb	@	rx.scala:64
yk	@	rx.scala:83
Pk.hb	@	Dogs.scala:16
(anonymous function)	@	mount.scala:76
Bo.j	@	AnonFunctions.scala:15
(anonymous function)	@	mount.scala:87
```

The culprit line is here: https://github.com/OlivierBlanvillain/monadic-html/blob/958184f2e81f042ece08981fe6043ed3c3c35012/monadic-rx/shared/src/main/scala/mhtml/rx.scala#L14

I'm surprised it even typechecks because I don't pass in a function into the map. When I pass a function (`.map(_ => <div>...</div>)`) it works.

**Comment by OlivierBlanvillain** [comment]
Bumped into this one already, here is the [root cause](https://github.com/scala/scala-xml/blob/892c2a7/src/main/scala/scala/xml/NodeSeq.scala#L55) :/

We might want to diverge from `scala-xml` to sanitize `scala.xml.Node` a bit. Because of the `Rx[_] → Node` implicit you can also get surprising behavior calling non existing `Rx` functions such as `fold`, which would trigger the implicit conversion and call the `scala.xml.Node#fold`...

**Comment by olafurpg** [comment]
Wow, that was unexpected. I find it crazy that all Seq have an `apply(i: Int): T`, since it's clearly a dangerous and non-idiomatic operation.

> We might want to diverge from scala-xml to sanitize scala.xml.Node a bit.

I think that sounds like a good plan. On one side, it's nice to piggy back on the existing implementation, but on the other hand it would be a huge improvement to clean out surprising behavior like this.

**Comment by OlivierBlanvillain** [comment]
It's a consequence of `Node <: Seq[Node]` :hankey:

We could solve this issue by marking every `Node` method we don't used `@deprecated` or something, but there is clearly room for investigation here which could also affect binary size.

**Comment by olafurpg** [comment]
Fixed in #26.
